corehq/form_processor/backends/sql/processor.py

```diff
@@ -105,6 +108,9 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None):
                 ledger_value.db for ledger_value in stock_result.models_to_save
             }
 
+        if any(has_tracked_vault_entries(on_model=form) for form in filter(None, processed_forms)):
```

Comment:
note to self: need to check if the vault entries need to be added first on the form to be tracked.

Reply:
confirmed that this is not needed. The concern was mainly in case of forms that failed to process and were getting "reprocessed", but that was okay because even in case of failures, the vault entries were already extracted out and were saved along with the form itself, which is done even in case of a failure.